components/inputs/input-radio-group.js

```diff
@@ -0,0 +1,239 @@
+		focusElem.focus();
+		setTimeout(() => focusElem.focus()); // timeout required when following link from form validation
+	}
+
+	#labelId = getUniqueId();
+
+	async #doUpdateChecked(newChecked, doFocus, doDispatchEvent) {
+		const radios = this.#getRadios();
+		let prevChecked = null;
+		radios.forEach(el => {
+			if (el._checked) prevChecked = el;
+		});
+		if (prevChecked === newChecked) return;
+
+		newChecked._checked = true;
+		if (prevChecked !== null) {
+			prevChecked._checked = false;
+		}
+		this.#recalculateState(true);
+
+		if (doDispatchEvent) {
+			this.dispatchEvent(new CustomEvent('change', {
+				bubbles: true,
+				composed: true,
+				detail: {
+					value: newChecked.value,
+					oldValue: prevChecked?.value
+				}
+			}));
+		}
+
+		if (doFocus) {
+			await newChecked.updateComplete; // wait for tabindex to be updated
+			newChecked.focus();
+		}
+	}
+
+	#getRadios() {
+		const elems = this.shadowRoot?.querySelector('slot')?.assignedElements();
+		if (!elems) return [];
+		return elems.filter(el => el.tagName === 'D2L-INPUT-RADIO');
+	}
+
+	#handleClick(e) {
+		if (e.target.tagName !== 'D2L-INPUT-RADIO') return;
+		if (e.target.disabled) return;
+		this.#doUpdateChecked(e.target, true, true);
+		e.preventDefault();
+	}
+
+	#handleKeyDown(e) {
+		if (e.target.tagName !== 'D2L-INPUT-RADIO') return;
+
+		const isRtl = (getComputedStyle(this).direction === 'rtl');
+		let newOffset = null;
+		if (e.key === ' ') {
+			newOffset = 0;
+		} else if (e.key === 'ArrowUp' || (!isRtl && e.key === 'ArrowLeft') || (isRtl && e.key === 'ArrowRight')) {
+			newOffset = -1;
+		} else if (e.key === 'ArrowDown' || (!isRtl && e.key === 'ArrowRight') || (isRtl && e.key === 'ArrowLeft')) {
+			newOffset = 1;
+		}
+
+		if (newOffset === null) return;
+
+		const radios = this.#getRadios().filter(el => !el.disabled || el._checked);
+		let checkedIndex = -1;
+		let firstFocusableIndex = -1;
+		radios.forEach((el, i) => {
+			if (el._checked) checkedIndex = i;
+			if (firstFocusableIndex < 0 && !el.disabled) firstFocusableIndex = i;
+		});
+		if (checkedIndex === -1) {
+			if (firstFocusableIndex === -1) return;
+			checkedIndex = firstFocusableIndex;
+		}
+
+		const newIndex = (checkedIndex + newOffset + radios.length) % radios.length;
+		this.#doUpdateChecked(radios[newIndex], true, true);
+
+		e.preventDefault();
+	}
+
+	#handleRadioChecked(e) {
+		if (e.detail.checked) {
+			this.#doUpdateChecked(e.target, false, false);
+		} else {
+			e.target._checked = false;
+			this.#recalculateState(true);
+		}
+	}
+
+	#handleSlotChange() {
+		this.#recalculateState(false);
+	}
+
+	#recalculateState(doValidate) {
+		const radios = this.#getRadios();
+		if (radios.length === 0) return;
+
+		let firstFocusable = null;
+		const checkedRadios = [];
+		radios.forEach(el => {
+			if (firstFocusable === null && !el.disabled) firstFocusable = el;
+			if (el._checked) checkedRadios.push(el);
+			el._isInitFromGroup = true;
```

Comment:
As soon as this gets set, the group is now managing the checked state of each radio.